### ISSUE TITLE
mbedls not working on Realtek RTL8195AM

### ISSUE THREAD

**Issue by gggekov** [ARMmbed/mbed-os-example-blinky#100]
`mbedls` and `mbed detect` don't work on the Realtek RTL8195AM board(https://os.mbed.com/platforms/Realtek-RTL8195AM/). 

I connect the board over USB, it appears as USB mass storage device but it's not recognised by `mbedls`. 
Here's the output that I get on Windows
```
`C:\Users\geogek01\Documents\mbedCLIproj\mbed-os-example-blinky [mbed-os-5.7.0-oob ≡ +1 ~1 -0 !]> mbedls

C:\Users\geogek01\Documents\mbedCLIproj\mbed-os-example-blinky [mbed-os-5.7.0-oob ≡ +1 ~1 -0 !]> mbed detect
[mbed] No mbed targets were detected on your system.`
C:\Users\geogek01\Documents\mbedCLIproj\mbed-os-example-blinky [mbed-os-5.7.0-oob ≡ +1 ~1 -0 !]> mbedls --version
1.2.14
```

How can I fix this issue? 

Thanks,
George

**Comment by MarceloSalazar** [comment]
This is what I get on OSX:

```
~/$ mbedls
WARNING:root:Could not read htm on from usb id None. Falling back to usb id
Traceback (most recent call last):
  File "/usr/local/bin/mbedls", line 11, in <module>
    load_entry_point('mbed-ls==1.3.4', 'console_scripts', 'mbedls')()
  File "/usr/local/lib/python2.7/site-packages/mbed_lstools/main.py", line 261, in mbedls_main
    ret_code = args.command(mbeds, args)
  File "/usr/local/lib/python2.7/site-packages/mbed_lstools/main.py", line 110, in print_table
    return print_mbeds(mbeds, args, False)
  File "/usr/local/lib/python2.7/site-packages/mbed_lstools/main.py", line 97, in print_mbeds
    devices = mbeds.list_mbeds(unique_names=True, read_details_txt=True)
  File "/usr/local/lib/python2.7/site-packages/mbed_lstools/lstools_base.py", line 147, in list_mbeds
    }[fs_interaction](device, platform_name_matcher)
  File "/usr/local/lib/python2.7/site-packages/mbed_lstools/lstools_base.py", line 183, in _fs_before_id_check
    self._update_device_from_htm(device)
  File "/usr/local/lib/python2.7/site-packages/mbed_lstools/lstools_base.py", line 218, in _update_device_from_htm
    device['platform_name'] = self.plat_db.get(device['target_id'][0:4])
TypeError: 'NoneType' object has no attribute '__getitem__'
```

```
~/mbed-os-example-blinky$ mbed detect
Traceback (most recent call last):
  File "/Users/marsal01/mbed/OOB/5.7/mbed-os-example-blinky/mbed-os/tools/detect_targets.py", line 76, in main
    muts = get_autodetected_MUTS_list()
  File "/Users/marsal01/mbed/OOB/5.7/mbed-os-example-blinky/mbed-os/tools/test_api.py", line 1663, in get_autodetected_MUTS_list
    return get_autodetected_MUTS(detect_muts_list, platform_name_filter=platform_name_filter)
  File "/Users/marsal01/mbed/OOB/5.7/mbed-os-example-blinky/mbed-os/tools/test_api.py", line 1689, in get_autodetected_MUTS
    'mcu_unique' : mut['platform_name_unique'] if 'platform_name_unique' in mut else "%s[%s]" % (mut['platform_name'], mut['target_id'][-4:]),
TypeError: 'NoneType' object has no attribute '__getitem__'
[ERROR] 'NoneType' object has no attribute '__getitem__'
[mbed] ERROR: "/usr/bin/python" returned error code 1.
[mbed] ERROR: Command "/usr/bin/python -u /Users/marsal01/mbed/OOB/5.7/mbed-os-example-blinky/mbed-os/tools/detect_targets.py" in "/Users/marsal01/mbed/OOB/5.7/mbed-os-example-blinky"
```

```
$ mbedls --version
1.3.4
```

@theotherjimmy @bridadan please have a look

**Comment by 0xc0170** [comment]
whats the version of realtek interface (daplink) ?

**Comment by bridadan** [comment]
@MarceloSalazar Could you also please run this with `mbedls --debug` and post the log? Thanks!

**Comment by gggekov** [comment]
I changed the version of the DAPLink firmware and now it works as expected. 
```
`C:\Users\geogek01\Documents\mbedCLIproj\mbed-os-example-blinky [mbed-os-5.7.0-oob ≡ +1 ~1 -0 !]> mbed detect

[mbed] Detected REALTEK_RTL8195AM, port COM8, mounted D:, interface version 0241:
[mbed] Supported toolchains for REALTEK_RTL8195AM
+-------------------+-----------+-----------+-----------+-----------+-----------+-----------+
| Target            | mbed OS 2 | mbed OS 5 |    ARM    |  GCC_ARM  |    IAR    |   ARMC6   |
+-------------------+-----------+-----------+-----------+-----------+-----------+-----------+
| REALTEK_RTL8195AM |     -     | Supported | Supported | Supported | Supported | Supported |
+-------------------+-----------+-----------+-----------+-----------+-----------+-----------+
Supported targets: 1
Supported toolchains: 4`
```
mbedls functions correctly on MAC as well.

**Comment by bridadan** [comment]
@gggekov Good to here that fixed your issue!

However, if you still have a realtek board with the old version of the firmware, could you please still give us the log output with `mbedls --debug`. mbedls shouldn't traceback no matter what you have plugged in and we'd like to fix that bug. Thanks!

**Comment by MarceloSalazar** [comment]
The problem goes away on both Windows and Mac when updating the interface FW.

However, for boards that come from factory with default FW, this is a problem.

See logs attached.

[log.txt](https://github.com/ARMmbed/mbed-os-example-blinky/files/1543236/log.txt)

**Comment by bridadan** [comment]
Ah perfect, thanks so much!

**Comment by bridadan** [comment]
@MarceloSalazar feel free to close this again

**Comment by adbridge** [comment]
@MarceloSalazar Can this be closed then ?

**Comment by MarceloSalazar** [comment]
Please let us verify that the fix in PR https://github.com/ARMmbed/mbed-ls/issues/276 solves this problem. Thanks!

**Comment by 0xc0170** [comment]
> Please let us verify that the fix in PR ARMmbed/mbed-ls#276 solves this problem. Thanks!

Resolved?
